### ISSUE TITLE
Consider stabilizing the Java agent extension mechanism?

### ISSUE THREAD

**Issue by trask** [open-telemetry/opentelemetry-java-instrumentation#8913]
Now that the `opentelemetry-sdk-extension-autoconfigure` module has been stabilized, what's remaining before stabilizing the Java agent extension mechanism?

(besides renaming `otel.javaagent.experimental.extensions` to `otel.javaagent.extensions`)

**Comment by mateuszrzeszutek** [comment]
> (besides renaming `otel.javaagent.experimental.extensions` to `otel.javaagent.extensions`)

That is already done actually, the `otel.javaagent.experimental.extensions` config is a leftover of a previous implementation I think.

Should we stabilize the `javaagent-extension-api` as well?

**Comment by trask** [comment]
> That is already done actually, the `otel.javaagent.experimental.extensions` config is a leftover of a previous implementation I think.

oh, yes, so it seems that the agent extension mechanism (at least when limited to sdk autoconfig) is already defacto stable

definitely worth revisiting stabilizing `javaagent-extension-api` sometime after #8999
